docs/apache-airflow-providers-openlineage/guides/developer.rst

```diff
@@ -50,4 +49,25 @@ Instead of returning complete OpenLineage event, the provider defines ``Operator
       run_facets: dict[str, BaseFacet] = Factory(dict)
       job_facets: dict[str, BaseFacet] = Factory(dict)
 
-OpenLineage integration takes care to enrich it with things like general Airflow facets, proper event time and type.
+OpenLineage integration itself takes care to enrich it with things like general Airflow facets, proper event time and type, creating proper OpenLineage RunEvent.
+
+How to properly implement OpenLineage methods?
+==============================================
+
+There are several things worth noting when implementing OpenLineage in operators.
+
+First, do not import OpenLineage methods on top-level, but in OL method itself.
+This allows users to use your provider even if they do not have OpenLineage provider installed.
+
```

Comment:
can we add an example here (or) link to an already existing example? The example will be useful for folks who are starting out on a new OL implementation